### PR TITLE
OpenAPI: Make sure we do not override defined parameters #4137

### LINKED ISSUE #4137: Overwriting parameters using annotation
**API Platform version(s) affected**: 2.6.2

**Description**  
When i try to overwrite parameters for an item operation using annotations, its added again in the swagger_docs.json instead of replacing the existing default one.

**How to reproduce**  
Here's how the ApiResource parameters annotation looks like:
```
 * @ApiResource(
 *     itemOperations={
 *         "get"={
 *             "openapi_context"={
 *                 "parameters"={
 *                     {
 *                         "name"="id",
 *                         "in"="path",
 *                         "required"=true,
 *                         "schema"={
 *                             "type"="string",
 *                             "format"="uuid"
 *                         }
 *                     }
 *                 }
 *             }
 *         }
 * )
 */
```

Here's how it looks after i generate the swagger_docs.json (parameters appear two times here):
```
                "parameters": [
                    {
                        "name": "id",
                        "in": "path",
                        "description": "",
                        "required": true,
                        "deprecated": false,
                        "schema": {
                            "type": "string",
                            "format": "uuid"
                        },
                        "style": "simple"
                    },
                    {
                        "name": "id",
                        "in": "path",
                        "description": "Resource identifier",
                        "required": true,
                        "deprecated": false,
                        "schema": {
                            "type": "string"
                        },
                        "style": "simple"
                    }
                ],
```

### PR DESCRIPTION
**Pull request by soyuka**
| Q             | A
| ------------- | ---
| Branch?       | 2.6
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #4137 #4124
| License       | MIT
| Doc PR        | na

Improves performances by 60% https://blackfire.io/profiles/compare/d6d1c64b-ec39-4ac9-ac9f-630dac371da4/graph

### CODE REVIEW

File: src/OpenApi/Factory/OpenApiFactory.php [hunk 1]
```diff
@@ -486,4 +511,25 @@ private function getSecuritySchemes(): array
 
         return $securitySchemes;
     }
+
+    private function appendSchemaDefinitions(\ArrayObject &$schemas, \ArrayObject $definitions): void
```
Comment (alanpoulain) [inline src/OpenApi/Factory/OpenApiFactory.php:515]: Is the reference necessary?

File: src/OpenApi/Factory/OpenApiFactory.php [hunk 2]
```diff
@@ -486,4 +511,25 @@ private function getSecuritySchemes(): array
...
+    {
+        foreach ($definitions as $key => $value) {
+            $schemas[$key] = $value;
+        }
+    }
+
+    /**
+     * @var Model\Parameter[]
```
Comment (alanpoulain) [inline src/OpenApi/Factory/OpenApiFactory.php:523]: Not needed?

File: src/OpenApi/Factory/OpenApiFactory.php (hunk 1 above)
Comment (soyuka) [reply src/OpenApi/Factory/OpenApiFactory.php:515]: I think it's more performant like this indeed

File: src/OpenApi/Factory/OpenApiFactory.php (hunk 2 above)
Comment (soyuka) [reply src/OpenApi/Factory/OpenApiFactory.php:523]: Yes or else I need to type `existingParameter` :)

**Review by alanpoulain — APPROVED** [review]

**Comment by alanpoulain** [comment]
Thank you!

File: src/OpenApi/Model/Parameter.php [hunk 3]
```diff
@@ -83,7 +83,7 @@ public function getDeprecated(): bool
         return $this->deprecated;
     }
 
-    public function canAllowEmptyValue(): bool
+    public function getAllowEmptyValue(): bool
```
Comment (guilliamxavier) [inline src/OpenApi/Model/Parameter.php:86]: BC break? Did you intend to duplicate (like the other ones below) rather than rename the existing?

**Pull request merged**